### DISCUSSION TITLE
v0.2.1+10

### DISCUSSION THREAD

**Discussion by hjiangsu** [thunder-app/thunder#167] [Announcements]
## Alpha Release [v0.2.1+10]

 Zoom zoom zoom, new update incoming and Happy Canada Day for my fellow Canadians out there (if there are any out there 👀) 🎉 

*I know I skipped v0.2.1+9 - this was because a build failed for TestFlight, which upped the version from v0.2.1+9 -> v0.2.1+10*

**Important: it is highly recommended to reinstall the app for the best (and least buggy) experience possible**

This update was mainly focused on improving overall performance for Thunder - you should hopefully see less lag as you scroll down your favourite feed or community! In addition to that, a few new major additions have been added, including the ability to view a user's profile, a revamped account page (for posts, comments, and saved posts), editing of comments, ability to download images and of course, more quality of life improvements to make your experience even better!

As always, this is an alpha release, so do expect bugs to pop up here and there. Without further ado, here are the changes:

### Major Additions
- Hopefully major improvements to performance within the app! You should feel that scrolling through the feed is smoother, and less laggy (hopefully) There is still an issue with some stuttering within long comment threads which will be looked at further
- Added ability to edit your own comments - editing will replace the reply swipe gesture if you are swiping on your own comment
- Added ability to download images - contribution from @MrAntonS
- Added back ability to collapse a comment, toggle-able in the settings
- Added pull to refresh on posts
- An additional button now shows up for each post in comfortable view to open the more actions modal
- Long pressing the post will bring up a modal to perform more actions such as visit community, user profile, sharing
- Added initial support for viewing user profiles - including their posts and comments
- Added top sort options - contribution from @JulianPaulus
- Upvoting and downvoting posts/comments now provides you with immediate feedback (no more waiting around)

### QoL Improvements
- Adjusted thickness of divider between posts to help differentiate posts
- Increased threshold for triggering a upvote/downvote on comments to hopefully have less accidental actions

### Fixes and Misc
- Fixed issue where you could not exit app when swiping back or using the back button
- Fixed issue where an error would be thrown if trying to access http pages - contribution from @vbh
- Fixed issue with missing haptic feedback on comment actions and other places
- Fixed issue where reply button would cover actions and comments for short posts
- Removed mark as read for mentions and replies that have already been read

And many more changes under the hood!

The main focuses for the next release are (but subject to change):
- Adding in a way to block users or communities
- Editing of posts
- Adding more options when creating posts
- Alternate ways of voting in comments outside of swipe gestures

----
If you find any issues or would like to suggest features, please open up a new issue! As always, contributions are always welcomed here.

<hr /><em>This discussion was created from the release <a href='https://github.com/hjiangsu/thunder/releases/tag/v0.2.1%2B10'>v0.2.1+10</a>.</em>

**Comment by Wailzy** [comment]
Excited for the new version :) 
Happy Canada Day!!

**Comment by MrAntonS** [comment]
This app starts to look fantastic! Great job everyone! Especially @hjiangsu 😄

**Comment by tom-james-watson** [comment]
The progress being made here is insane. By far the slickest Lemmy experience out there at the moment.

**Reply by hjiangsu** [reply]
Thanks! It'll be constant improvements from now on hopefully _crosses fingers_

**Comment by krestenlaust** [comment]
Is reinstallation also recommended for local apk installation?

**Reply by hjiangsu** [reply]
Generally, I would say to reinstall if at all possible- this is because it's still an alpha release, and so there's a lot of things changing under the hood constantly

Once it becomes more stable and there's less changes to the underlying architecture, then that would be the time where reinstall shouldn't be necessary!

**Reply by tom-james-watson** [reply]
What does "reinstall" actually mean? To uninstall the current app before installing the new APK?

**Reply by hjiangsu** [reply]
You can either uninstall and install the app (more ideal), or clear the app data and cache!

**Reply by tom-james-watson** [reply]
By the way when I did and uninstall and install of the fresh APK, that didn't actually clear the app data. I think if the issue is around non forwards-compatible state between alpha versions then you need to explicitly say to clear the app data/cache.
